openedx/core/djangoapps/credit/signals.py

```diff
@@ -18,3 +21,27 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
     from .tasks import update_credit_course_requirements
 
     update_credit_course_requirements.delay(unicode(course_key))
+
+
+@receiver(MIN_GRADE_REQUIREMENT_STATUS)
+def listen_for_grade_calculation(sender, request, grade_summary, course_key, deadline, **kwargs):  # pylint: disable=unused-argument
+    """Receive 'MIN_GRADE_REQUIREMENT_STATUS' signal and update minimum grade
+    requirement status.
+    """
+    from openedx.core.djangoapps.credit.models import CreditCourse, CreditRequirementStatus, CreditRequirement
```

Comment:
Please use the functions in `credit.api` instead of calling these models directly.  There should be API functions available to do everything you need here.